### PR TITLE
feat(sidenav): migrate to core tokens

### PR DESCRIPTION
**Pull request by pfulton**
## Description

Migration for SideNav. Has been merged and released by Spectrum CSS

## Related issue(s)

-

## How has this been tested?

-   [ ] _Test case 1_
    1. Go here
    2. Do this
-   [ ] _Test case 2_
    1. Go here
    2. Do this

## Types of changes

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [ ] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist

-   [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [ ] My code follows the code style of this project.
-   [ ] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [ ] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [ ] I have added tests to cover my changes.
-   [ ] All new and existing tests passed.
-   [ ] I have reviewed at the Accessibility Practices for this feature, see: [Aria Practices](https://www.w3.org/TR/wai-aria-practices/)

## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.

### CODE REVIEW

File: packages/sidenav/src/spectrum-sidenav-heading.css [hunk 1]
```diff
@@ -12,68 +12,156 @@ governing permissions and limitations under the License.
...
-        var(--spectrum-global-dimension-size-50)
+        --mod-sidenav-header-line-height,
+        var(--spectrum-sidenav-header-line-height)
     );
-    margin-top: var(
-        --spectrum-sidenav-heading-gap-top,
-        var(--spectrum-global-dimension-size-200)
+    margin-block-end: var(--spectrum-sidenav-heading-bottom-margin);
+    margin-block-start: calc(
+        var(--spectrum-sidenav-heading-top-margin) - var(--spectrum-sidenav-gap)
```
[the remark this answers is not in the document]
Comment (najikahalsema) [reply packages/sidenav/src/spectrum-sidenav-heading.css:156]: that's right.

[the remark this answers is not in the document]
Comment (jenndiaz) [reply packages/sidenav/src/spectrum-sidenav-heading.css:156]: I rebased and before the pr most recent changes I have are `docs: fixed typos in documentation readme` which was from just this morning. Still not getting the custom tokens.

[the remark this answers is not in the document]
Comment (najikahalsema) [reply packages/sidenav/src/spectrum-sidenav-heading.css:156]: are you only not seeing them on the docs site, or are you getting the changes elsewhere?

[the remark this answers is not in the document]
Comment (jenndiaz) [reply packages/sidenav/src/spectrum-sidenav-heading.css:156]: I am not seeing these values in `tools/styles/tokens/spectrum/custom-medium-vars.css` where on previous components this generated without me needing to do anything different.

 so in the browser for the docs site  as well as in the storybook, they are just not equating to anything.

[the remark this answers is not in the document]
Comment (Westbrook) [reply packages/sidenav/src/spectrum-sidenav-heading.css:156]: Are these values form a newer `@spectrum-css/tokens` package than is available in this branch?

[the remark this answers is not in the document]
Comment (jenndiaz) [reply packages/sidenav/src/spectrum-sidenav-heading.css:156]: yes, that is exactly why.  Sorry for going down the wrong path, Thank you both for looking.

**Comment by jenndiaz** [comment]
Not sure if this is a concern, the new designs for sidenav did set a width and the docs site implementation the selected gray background and key focus is now cut off. 
Screenshot 2023-07-18 at 4 19 10 PM

We could fix this by adding `style="--mod-sidenav-width: 177px;"` to match the current width

**Comment by Westbrook** [comment]
Could we apply `--mod-sidenav-width: 100%` and see if that solves for both scales?

**Comment by Westbrook** [comment]
In the docs site, when you are on https://migrate-side-nav-tokens--spectrum-web-components.netlify.app/components/sidenav/ and you click and hold down on any other Side Nav item (e.g. `Picker`) the content shifts up for some reason...

**Comment by Westbrook** [comment]
> Not sure if this is a concern, the new designs for sidenav did set a width and the docs site implementation the selected gray background and key focus is now cut off. Screenshot 2023-07-18 at 4 19 10 PM
> 
> We could fix this by adding `style="--mod-sidenav-width: 177px;"` to match the current width

This API feels a little weird. I _should_ be able to apply a width to the `<sp-sidenav>` and expext it to cascade to the rest of the pattern layout.

This can be achieved via `--mod-sidenav-width: 100%` so I wonder if that could just be the default and the width, or this custom property, apply to the root element rather than something somewhat random within the larger pattern?

**Comment by jenndiaz** [comment]
> In the docs site, when you are on https://migrate-side-nav-tokens--spectrum-web-components.netlify.app/components/sidenav/ and you click and hold down on any other Side Nav item (e.g. `Picker`) the content shifts up for some reason...

This was coming from margin added `projects/documentation/src/components/styles.css` 
```
sp-sidenav-item:not([tabindex]) {
    display: list-item;
    list-style-type: none;
    margin: var(
            --spectrum-sidenav-item-gap,
            var(--spectrum-global-dimension-size-50)
        )
        0;
}
```
From what I can tell you all are toggling between `focusable` and `tabindex="-1"` and were adding margin between the two which was causing the jumping. I removed the margin and it has gone away. There is a bit of a jump on page load but this was already existing. 

I pushed up a commit to remove the margin from this class, with the new component the gap between side nav items is handled by ` margin-block-end: var(--mod-sidenav-gap, var(--spectrum-sidenav-gap));`  on each item.

File: projects/documentation/src/components/styles.css [hunk 2]
```diff
@@ -280,15 +280,6 @@ sp-sidenav:not(:defined) {
...
     list-style-type: none;
-    margin: var(
-            --spectrum-sidenav-item-gap,
-            var(--spectrum-global-dimension-size-50)
-        )
-        0;
-}
-
-docs-page > sp-sidenav {
-    content-visibility: auto;
```
Comment (jenndiaz) [inline projects/documentation/src/components/styles.css:291]: @Westbrook any concerns about removing `content-visibility: auto;` ? 

With updating the width to 100% this was cutting off the focus-ring, removing it solved it. But I do not have full context as to why it was being used previously. 

Of note the property is [experimental and not currently compatible with safari.](https://developer.mozilla.org/en-US/docs/Web/CSS/content-visibility)

Comment (jenndiaz) [reply projects/documentation/src/components/styles.css:291]: @pfulton just so you have this info as well! ^^

**Comment by jenndiaz** [comment]
@Westbrook Addressed all the feedback you had and the couple of things I found. VRT failures are all expected based on design updates. Would you like to give it another look before we do our full release?

**Review by Westbrook — COMMENTED** [review]
LGTM! Once we've got a stable release from CSS, we'll be on out way with this one. 🙇🏼 Great work.

**Comment by jenndiaz** [comment]
This has been merged and released in Spectrum CSS, and the full release is: @spectrum-css/sidenav@4.0.0

**Review by Westbrook — APPROVED** [review]
@jenndiaz thanks for seeing this through! LGTM. 🚀

**Pull request merged**